### PR TITLE
Yaml Game Config

### PR DESCRIPTION
**Pull request by alecpm**
This is intended as a starting point for feature development. Adds a sample game configuration.

The when `exp.configure()` is run, the experiment loads and stores the full game configuration object in `exp.game_config`. It stores all objects indexed by `object_id` in `exp.object_config`, and all transitions indexed by `(actor_start , target_start )` tuples in `exp.transition_config`.

### CODE REVIEW

**Comment by alecpm** [comment]
There's a problem here, which is that `dallinger.utils.check_experiment_dependencies` uses the `requirements.txt` file (which includes `PyYAML`) and the packages there using `importlib.util.find_spec`, but that method can't see `PyYAML` only the importable name `yaml`.

This will require a small fix in Dallinger, which should be helpful for other dependencies whose package name differs from its importable name.

**Comment by alecpm** [comment]
I've added a PR for the Dallinger fix here: https://github.com/Dallinger/Dallinger/pull/5409

File: setup.py [hunk 1]
```diff
@@ -74,13 +70,13 @@
...
             'codecov',
             'flake8',
-            'google-compute-engine',
             'pytest',
             'recommonmark',
             'Sphinx',
             'sphinxcontrib-spelling',
             'tox',
             'mock',
+            'pip-tools',
```
Comment (silviot) [inline setup.py:79]: Thanks for moving this to the extras!

**Review by silviot — APPROVED** [review]

File: dlgr/griduniverse/experiment.py [hunk 2]
```diff
@@ -1140,6 +1145,16 @@ def configure(self):
                                                         self.num_participants)
         self.network_factory = self.config.get('network', 'FullyConnected')
 
+        game_config_file = os.path.join(os.getcwd(), GAME_CONFIG_FILE)
```
Comment (jessesnyder) [inline dlgr/griduniverse/experiment.py:1148]: Total nit: we've been moving toward Pathlib, and it seems like we should use it unless there's a reason not to(?)

**Review by jessesnyder — APPROVED** [review]
Looks great! One nit about file path handling.

File: dlgr/griduniverse/experiment.py (hunk 2 above)
Comment (alecpm) [reply dlgr/griduniverse/experiment.py:1148]: @jessesnyder Sounds good. I trust you to make that update. I'm not sure if I've ever used `Pathlib`

**Pull request merged**